src/math/crypto/curve/ec_point.zig

```diff
@@ -0,0 +1,149 @@
+        const w = t.mul(t).sub(Felt252.two().mul(v));
+
+        const uy = u.mul(self.y);
+
+        self.* = .{
+            .x = u.mul(w),
+            .y = t.mul(v.sub(w)).sub(Felt252.two().mul(uy).mul(uy)),
+            .z = u.mul(u).mul(u),
+            .infinity = self.infinity,
+        };
+    }
+
+    pub fn addAssignAffinePoint(self: *Self, rhs: AffinePoint) void {
+        if (rhs.infinity) {
+            return;
+        }
+
+        if (self.infinity) {
+            self.x = rhs.x;
+            self.y = rhs.y;
+            self.z = Felt252.one();
+            self.infinity = rhs.infinity;
+            return;
+        }
+
+        const u_0 = self.x;
+        const u_1 = rhs.x.mul(self.z);
+        const t0 = self.y;
+        const t1 = rhs.y.mul(self.z);
+
+        if (u_0.equal(u_1)) {
+            if (!t0.equal(t1)) {
+                self.infinity = true;
+                return;
+            } else {
+                self.doubleAssign();
+                return;
+            }
+        }
+
+        const t = t0.sub(t1);
+        const u = u_0.sub(u_1);
+        const u_2 = u.mul(u);
+
+        const v = self.z;
+        const w = t.mul(t).mul(v).sub(u_2.mul(u_0.add(u_1)));
+        const u_3 = u.mul(u_2);
+
+        const x = u.mul(w);
+        const y = t.mul(u_0.mul(u_2).sub(w)).sub(t0.mul(u_3));
+        const z = u_3.mul(v);
+
+        self.x = x;
+        self.y = y;
+        self.z = z;
+    }
+};
+
+pub const AffinePoint = struct {
+    const Self = @This();
+    x: Felt252,
+    y: Felt252,
+    infinity: bool,
+
+    pub fn addAssign(self: *Self, rhs: *AffinePoint) void {
+        if (rhs.infinity) {
+            return;
+        }
+        if (self.infinity) {
+            self.x = rhs.x;
+            self.y = rhs.y;
+            self.infinity = rhs.infinity;
+            return;
+        }
+
+        if (self.x.equal(rhs.x)) {
+            if (self.y.equal(rhs.y.neg())) {
+                self.x = Felt252.zero();
+                self.y = Felt252.zero();
+                self.infinity = true;
+                return;
+            }
+            self.doubleAssign();
+            return;
+        }
+
+        // l = (y2-y1)/(x2-x1)
+        const lambda = rhs.y.sub(self.y).mul(rhs.x.sub(self.x).inv().?);
+
+        const result_x = lambda.mul(lambda).sub(self.x).sub(rhs.x);
+        self.y = lambda.mul(self.x.sub(result_x)).sub(self.y);
+        self.x = result_x;
+    }
+
+    pub fn doubleAssign(self: *Self) void {
+        if (self.infinity) {
+            return;
+        }
+
+        // l = (3x^2+a)/2y with a=1 from stark curve
+        const lambda = Felt252.three().mul(self.x.mul(self.x)).add(Felt252.one()).mul(Felt252.two().mul(self.y).inv().?);
+
+        const result_x = lambda.mul(lambda).sub(self.x).sub(self.x);
+        self.y = lambda.mul(self.x.sub(result_x)).sub(self.y);
+        self.x = result_x;
+    }
+
+    pub fn fromProjectivePoint(p: *ProjectivePoint) Self {
+        const zinv = p.z.inv().?;
```

Comment:
Here you will have an error you need to handle in the case `p.z.inv()` is null because you will try to access a null field, returning a panic.

Reply:
as i understand it should not be null (as rust implementation they unwrap also, expecting not null), should we handle null value?

Reply:
I think so just in case, maybe throwing an error or something like that to avoid panic, wdt?

Reply:
maybe use some like return unreachble?

Reply:
maybe smth like `if (p.z.inv) |zinv| { doStuff } else { ... } ` , maybe `unreachable` is actually best if you're sure there is never a null value there?